### PR TITLE
[stable/mongodb] Introduce extra volumes for secondary stateful set

### PR DESCRIPTION
**Pull request by sventschui**
#### What this PR does / why we need it:
Adds the ability to add extra volumes / mounts / claim templates to the secondary nodes. We want to use this to mount a PV to the secondaries where we will temporarily store backups. As our cluster does not support RWX storage we can not attach one PV to primary+secondary stateful sets/pods.

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[stable/mychartname]`)

### CODE REVIEW

**Comment by javsalgar** [comment]
Hi,

Thanks for the PR. Could you please bump the minor version? It is adding a non-breaking feature

File: stable/mongodb/templates/statefulset-secondary-rs.yaml [hunk 1]
```diff
@@ -195,6 +195,9 @@ spec:
           {{- if .Values.extraVolumeMounts }}
 {{ toYaml .Values.extraVolumeMounts | indent 12}}
           {{- end }}
+          {{- if .Values.extraVolumeMountsSecondary }}
+{{ toYaml .Values.extraVolumeMountsSecondary | indent 12}}
+{{- end }}
```
Comment (javsalgar) [inline stable/mongodb/templates/statefulset-secondary-rs.yaml:200]: Indent the `{{-end }}`

File: stable/mongodb/templates/statefulset-secondary-rs.yaml [hunk 2]
```diff
@@ -195,6 +195,9 @@ spec:
           {{- if .Values.extraVolumeMounts }}
 {{ toYaml .Values.extraVolumeMounts | indent 12}}
           {{- end }}
+          {{- if .Values.extraVolumeMountsSecondary }}
+{{ toYaml .Values.extraVolumeMountsSecondary | indent 12}}
```
Comment (javsalgar) [inline stable/mongodb/templates/statefulset-secondary-rs.yaml:199]: Add a space after 12

File: stable/mongodb/templates/statefulset-secondary-rs.yaml [hunk 3]
```diff
@@ -257,8 +260,16 @@ spec:
...
 {{ toYaml .Values.extraVolumes | indent 8}}
         {{- end }}
-{{- if .Values.persistence.enabled }}
+        {{- if .Values.extraVolumesSecondary }}
+{{ toYaml .Values.extraVolumesSecondary | indent 8}}
+        {{- end }}
+{{- if not .Values.persistence.enabled }}
+        - name: datadir
+          emptyDir: {}
+{{- end}}
```
Comment (javsalgar) [inline stable/mongodb/templates/statefulset-secondary-rs.yaml:269]: Add extra space after `end`

Comment (javsalgar) [reply stable/mongodb/templates/statefulset-secondary-rs.yaml:269]: Also indent the `if` and `end` like the rest

**Review by javsalgar — CHANGES_REQUESTED** [review]
Thanks! Minor changes requested

**Comment by sventschui** [comment]
Closing as the use case is no longer applicable, this chart moved to another repo and I currently do not have the capacity to apply the requested changes

**Pull request closed without merging**